x-pack/legacy/plugins/code/server/indexer/lsp_incremental_indexer.ts

```diff
@@ -123,6 +123,7 @@ export class LspIncrementalIndexer extends LspIndexer {
   protected async *getIndexRequestIterator(): AsyncIterableIterator<LspIncIndexRequest> {
     try {
       if (this.diff) {
+        await this.prepareWorkspace();
```

Comment:
why do we need to `prepareWorkspace()` here? it's assuming that the `this.diff` should have already been populated. Maybe move to the place where the `this.diff` has been populated?